### PR TITLE
Ocean Waves improvements

### PR DESCRIPTION
**Pull request by mbkuhn**
## Summary

Collection of improvements for the Ocean Waves functionality.

## Pull request type

- [x] Bugfix
- [x] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [x] Documentation content changes
- [ ] Other (please describe):

## Checklist

The following is included:

- [ ] new unit-test(s)
- [x] new regression test(s)
- [x] documentation for new capability

This PR was tested by running:

- the unit tests
  - [ ] on GPU 
  - [x] on CPU 
- the regression tests
  - [ ] on GPU 
  - [x] on CPU

### CODE REVIEW

**Comment by mbkuhn** [comment]
diffs (as expected) introduced in ow_linear, ow_stokes, and abl_multiphase_laminar. relatively minor diffs, except for ow_stokes because the domain has changed.

File: amr-wind/ocean_waves/relaxation_zones/relaxation_zones_ops.cpp [hunk 1]
```diff
@@ -33,6 +33,7 @@ void read_inputs(
     } else {
         pp.query("numerical_beach_length", wdata.beach_length);
         wdata.has_beach = true;
+        pp.query("numerical_beach_lfactor", wdata.beach_length_factor);
```
Comment (marchdf) [inline amr-wind/ocean_waves/relaxation_zones/relaxation_zones_ops.cpp:36]: more descriptive user facing option? `numerical_beach_length_factor` ?

File: amr-wind/ocean_waves/relaxation_zones/stokes_waves_ops.H [hunk 2]
```diff
@@ -88,16 +88,48 @@ struct InitDataOp<StokesWaves>
             const auto& phi = m_levelset(level).array(mfi);
             const auto& vel = velocity(level).array(mfi);
 
-            const amrex::Real zsl = wdata.zsl;
+            const amrex::Real waveheight = wdata.wave_height;
```
Comment (marchdf) [inline amr-wind/ocean_waves/relaxation_zones/stokes_waves_ops.H:91]: for these 3, maybe for consistency with the rest: `wave_*` / `water_*`?

**Review by marchdf — CHANGES_REQUESTED** [review]

**Comment by marchdf** [comment]
Test pass for me on CPU (other than the ones you noted).

**Review by marchdf — APPROVED** [review]

**Pull request merged**